File: leetcode-162_findPeakElement.js
//查找数组的峰值
//给定一数组, 相邻元素不等, 查找任一峰值
//如果后一元素比前面的大, 说明在上山, 直到后面元素比前一个小, 说明在下山, 那么前一个元素就是峰顶
function findPeakElement(nums)
{
    for (i = 1; i < nums.length; i++) {
        if (nums[i] < nums[i - 1]) {
            return i - 1;
        }
    }
    return nums.length - 1;
}

var arr = [1, 2, 3];
console.log(findPeakElement(arr));

File: leetcode-162_find-peak-element.java
/**
 * 给定一个 int 数组，返回一个峰顶元素的下标
 * 峰顶元素即 nums[i - 1] < nums[i] > nums[i + 1] 中的 nums[i]
 */
public int findPeakElement(int[] nums) {
    for (int i = 1; i < nums.length; i++) {
        if (nums[i] < nums[i - 1]) {
            return i - 1;
        }
    }
    // 如果元素一直在增大，则最后一个元素即峰顶
    return nums.length - 1;
}
